src/Agda2Hs/Compile/Type.hs

```diff
@@ -33,68 +34,52 @@ import Agda.Utils.Size ( Sized(size) )
 import Agda.Utils.Functor ( ($>) )
 
 import Agda2Hs.Compile.Name ( compileQName )
-import Agda2Hs.Compile.Term ( compileVar )
 import Agda2Hs.Compile.Types
 import Agda2Hs.Compile.Utils
 import Agda2Hs.AgdaUtils
 import Agda2Hs.HsUtils
 
-isSpecialType :: QName -> Maybe (QName -> Elims -> C (Hs.Type ()))
-isSpecialType = prettyShow >>> \case
-  "Haskell.Prim.Tuple._×_"   -> Just tupleType
-  "Haskell.Prim.Tuple._×_×_" -> Just tupleType
-  "Haskell.Extra.Erase.Erase" -> Just erasedType
+
+-- | Type definitions from the prelude that get special translation rules.
+isSpecialType :: QName -> Maybe (Elims -> C (Hs.Type ()))
+isSpecialType x = case prettyShow x of
```

Comment:
There is no need for the intermediate variable `x` here, except for readability right?

I personally find the previous point-free phrasing nicer, but if this was intentional the import of `Arrow.>>>` becomes obsolete I think.